utils/build-script-impl

```diff
@@ -1929,20 +1938,26 @@ for deployment_target in "${HOST_TARGET}" "${CROSS_TOOLS_DEPLOYMENT_TARGETS[@]}"
                 ;;
             xctest)
                 SWIFTC_BIN="$(build_directory_bin ${deployment_target} swift)/swiftc"
-                XCTEST_BUILD_DIR=$(build_directory ${deployment_target} xctest)
                 if [[ "$(uname -s)" == "Darwin" ]] ; then
+                    # xcodebuild requires swift-stdlib-tool to build a Swift
+                    # framework. This is normally present when building XCTest
+                    # via a packaged .xctoolchain, but here we are using the
+                    # swiftc that was just built--no toolchain exists yet. As a
+                    # result, we must copy swift-stdlib-tool ourselves.
+                    copy_swift_stdlib_tool_substitute "$(build_directory_bin ${deployment_target} swift)/swift-stdlib-tool"
+
+                    # The Linux build places build products in the
+                    # XCTEST_BUILD_DIR. This, and installation of XCTest in
+                    # general, is not supported on OS X.
                     set -x
                     xcodebuild \
                         -workspace "${XCTEST_SOURCE_DIR}"/XCTest.xcworkspace \
                         -scheme SwiftXCTest \
                         SWIFT_EXEC="${SWIFTC_BIN}" \
-                        SWIFT_LINK_OBJC_RUNTIME=YES \
-                        SKIP_INSTALL=NO \
-                        DEPLOYMENT_LOCATION=YES \
-                        DSTROOT="${XCTEST_BUILD_DIR}" \
```

Comment:
The idea here is that the Swift build script no longer minds where the build products are placed. I'd like for the build script to be able to build and test XCTest to make it easier for newcomers to contribute (everyone uses the same invocation to test). But attempting to have it install build products in a specific location was a misstep on my part: we don't support installing corelibs-xctest on OS X, so we shouldn't try to place build products in a specific directory either.

With this change, build products are placed in the usual Xcode location: the derived data folder.

Alternatively, it might be possible to respect `DSTROOT` here, but I just can't get the build to work when doing so... :sweat_smile: